diagnostic.js

```diff
@@ -30,25 +30,68 @@ mongoose.connect('mongodb://localhost/mongoose-crud');
 
 /// ADD YOUR CODE BELOW
 
-const create = (name, description, startYear, endYear) => {};
+'use strict';
+
+const models = require('app/models');
+const Movement = models.movement;
+
+const create = (name, description, startYear, endYear) => {
+  Movement.create(name, description, startYear, endYear)
+    .then(movement => {
+      return movement.save();
+    })
+    .then(movement => print(movement.json()))
+    .catch(err => console.error(err));
+};
 // Success -> Print new Movement as JSON
 // Failure -> Console.error
 
-const index = () => {};
+const index = () => {
+  Movement.find()
+    .then(movements => print(movements.json())
+    .catch(err => console.error(err));
+};
 // Success -> Print all Movements as JSON
 // Failure -> Console.error
 
-const show = (id) => {};
+const show = (id) => {
+  Movement.findById(id)
+    .then(movement => movement ? print(movements.json() : next())  // If you find an book, send the JSON,
+    .catch(err => next(err));                          // otherwise go to the next action
+};
 // Success -> If the specified Movement exists, print it as JSON;
 //              otherwise, print "Not Found" and exit.
 // Failure -> Console.error
 
-const update = (id, field, value) => {};
+const update = (id, field, value) => {
+  let movement = {_id: id};
+  Movement.findOneAndUpdate(movement, field, value)
+  .then(movement => {
+    if (!movement) {
+      return next();
+    }
+
+    return movement.update(movement, field, value)
```

Comment:
You've already updated the movement. Additionally, `update` doesn't return the newly modified document, so you can't pass it to the `then` promise.